source/Halibut/Transport/Protocol/MessageSerializer.cs

```diff
@@ -75,10 +90,11 @@ T ReadCompressedMessageRewindable<T>(Stream stream, IRewindableBuffer rewindable
             try
             {
                 using (var zip = new DeflateStream(stream, CompressionMode.Decompress, true))
-                using (var bson = new BsonDataReader(zip) { CloseInput = false })
+                using (var decompressedObservableStream = new ObservableStream(zip))
```

Comment:
Interesting that the `ObservableStream` does not need to be told to not dispose the inner stream. I guess it works because we tell the `DeflateStream` not to dispose its inner stream.

Reply:
Luke and I had a chat about this, and decided it was a good idea to pass in a flag in the constructor to choose what happens to the underlying stream on dispose.

This will make it clear what is going to happen to the reader/future author.